src/server/endpoints/covidcast_utils/smooth_diff.py

```diff
@@ -0,0 +1,168 @@
+from collections import ChainMap
+from datetime import date
+from itertools import groupby, chain
+from numbers import Number
+from typing import Iterable, Dict, List, Optional, Union
+
+from more_itertools import windowed, peekable
+from pandas import date_range, Timedelta
+from numpy import nan, nan_to_num, array, dot, isnan
+
+
+def _smoother(values: List[Number], kernel: Optional[List[Number]] = None) -> Number:
+    """Basic smoother.
+
+    If kernel passed, uses the kernel as summation weights. If something is wrong,
+    defaults to the mean.
+    """
+    try:
+        if kernel:
+            kernel = array(kernel, copy=False)
+        else:
+            raise ValueError
+        values = array(values, copy=False)
+        smoothed_value = dot(values, kernel)
+    except (ValueError, TypeError):
+        smoothed_value = array(values).mean()
+
+    return round(smoothed_value, 8)
+
+
+def _iso_int_to_date(di: int) -> date:
```

Comment:
isn't there a utility function for that?

Reply:
All I could find was Python's [datetime.fromisoformat](https://docs.python.org/3/library/datetime.html?highlight=datetime#datetime.date.fromisoformat) but it expects dashes in the string. I saw you wrote some time parsers, but they only converted between str and int.

Reply:
what about https://github.com/cmu-delphi/delphi-epidata/blob/dev/src/server/utils/dates.py

Reply:
ah i didnt see these, thanks